src/environments.js

```diff
@@ -1,326 +1,16 @@
 // @flow
-import ParseNode from "./ParseNode";
-import ParseError from "./ParseError";
-
-import type Parser from "./Parser";
-import type {ArgType, Mode, StyleStr} from "./types";
-
-/**
- * The context contains the following properties:
- *  - mode: current parsing mode.
- *  - envName: the name of the environment, one of the listed names.
- *  - parser: the parser object.
- */
-type EnvContext = {
-    mode: Mode,
-    envName: string,
-    parser: Parser,
-};
-
-/**
- * The handler function receives two arguments
- *  - context: information and references provided by the parser
- *  - args: an array of arguments passed to \begin{name}
- */
-type EnvHandler = (context: EnvContext, args: ParseNode[]) => ParseNode;
-
-/**
- *  - numArgs: (default 0) The number of arguments after the \begin{name} function.
- *  - argTypes: (optional) Just like for a function
- *  - allowedInText: (default false) Whether or not the environment is allowed
- *                   inside text mode (not enforced yet).
- *  - numOptionalArgs: (default 0) Just like for a function
- */
-type EnvProps = {
-    numArgs?: number,
-    argTypes?: ArgType[],
-    allowedInText?: boolean,
-    numOptionalArgs?: number,
-};
-
-type EnvData = {
-    numArgs: number,
-    argTypes?: ArgType[],
-    greediness: number,
-    allowedInText: boolean,
-    numOptionalArgs: number,
-    handler: EnvHandler,
-};
-const environments: {[string]: EnvData} = {};
-export default environments;
-
-// Data stored in the ParseNode associated with the environment.
-type AlignSpec = { type: "separator", separator: string } | {
-    type: "align",
-    align: string,
-    pregap?: number,
-    postgap?: number,
+import {_environments} from "./defineEnvironment";
+
+const environments = {
```

Comment:
Cool! Is the motivation behind this abstraction to limit access to the `_environments` object? Or do you expect for the bodies of `has` and `get` to eventually become more complicated?

If this API was added solely for clarity, I don't think `has` and `get` improve readability over `hasOwnProperty` and `[]`.

Reply:
The motivation was to limit access.  The plan is to do the same for `_functions` and `_symbols`.  Not sure how necessary it is.